reporter/datadog_reporter.go

```diff
@@ -355,6 +359,139 @@ func (r *DatadogReporter) reportProfile(ctx context.Context) error {
 	return err
 }
 
+func (r *DatadogReporter) processSample(sample *pprofile.Sample, profile *pprofile.Profile,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** ‏ Not sure extracting this code into a function is still useful given the small amount of changes in it (and it makes reviewing the actual changes harder).

Reply:
I would also vote for keeping this in the same function for now, or at least splitting it in two commits if possible, to make this easier to review 🙏

Reply:
I will do a separate PR then, I was thinking of squash merging this PR